src/wallet/rpcwallet.cpp

```diff
@@ -1729,11 +1735,34 @@ UniValue listsinceblock(const JSONRPCRequest& request)
             ListTransactions(tx, "*", 0, true, transactions, filter);
     }
 
+    // when a reorg'd block is requested, we also list any relevant transactions
+    // in the blocks of the chain that was detached
+    UniValue reorged(UniValue::VARR);
+    while (paltindex && paltindex != pindex)
+    {
+        CBlock block;
+        if (!ReadBlockFromDisk(block, paltindex, Params().GetConsensus()))
+        {
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Can't read block from disk");
+        }
+        for (const CTransactionRef& tx : block.vtx)
+        {
+            if (pwalletMain->mapWallet.count(tx->GetHash()))
+            {
+                // Use -depth as minDepth parameter to ListTransactions, as these transactions have
+                // negative (at minimum -depth) confirmations.
```

Comment:
Would remove parentheses since (at minimum -depth) is the important part. Also, technically I believe the minimum is 1-depth. Maybe:

```
// Use -depth as minDepth parameter to ListTransactions to prevent incoming
// transactions from being filtered. These transactions have negative
// confirmations, but always greater than -depth.
```